editor/src/document/document_file.rs

```diff
@@ -372,12 +372,29 @@ impl MessageHandler<DocumentMessage, &InputPreprocessor> for DocumentMessageHand
 				Err(e) => log::error!("DocumentError: {:?}", e),
 				Ok(_) => (),
 			},
-			RenderDocument => responses.push_back(
+			RenderDocument => responses.extend([
 				FrontendMessage::UpdateCanvas {
 					document: self.document.render_root(),
 				}
 				.into(),
-			),
+				FrontendMessage::UpdateScrollbars {
```

Comment:
I don't like the functionality split between frontend and backend, either send the correct unmodified viewport bounds to the frontend, or just send {xpos, xscale, ypos, yscale} for the scrollbars. I think I would prefer the second approach

Reply:
Done